rust/src/smb/ntlmssp_records.rs

```diff
@@ -101,9 +101,9 @@ pub fn parse_ntlm_auth_record(i: &[u8]) -> IResult<&[u8], NTLMSSPAuthRecord> {
 
     // subtrack 12 as idenfier (8) and type (4) are cut before we are called
     // subtract 60 for the len/offset/maxlen fields above
-    let (i, _) = cond(nego_flags.1==1, |b| take(domain_blob_offset - (12 + 60))(b))(i)?;
+    let (i, _) = cond(nego_flags.1==1 && domain_blob_offset > 72, |b| take(domain_blob_offset - (12 + 60))(b))(i)?;
```

Comment:
should the `domain_blob_offset > 72` check move to `let (i, domain_blob_offset) = le_u32(i)?;` as a `verify!` check?

Reply:
I do not think so:

1 because it depends on `nego_flags.1==1`
2 For SMB, there can be dummy values such as zero for these fields, when they get unused